### DISCUSSION TITLE
How to decrease gatsby build time?

### DISCUSSION THREAD

**Discussion by snussik** [gatsbyjs/gatsby#28952] [Help]
In general, its obvious that more memory and cpu you have on instance, the quicker builds you get. But I made tests on my quite big data source: 5000 rows in a plain Postgres table on different aws  instances and saw no performance increase:
- m5.xlarge (vCPU/Memory): 4/16: **27m 27s**
- c4.2xlarge (vCPU/Memory): 8/15: **23m 6s**
- c4.4xlarge (vCPU/Memory): 16/30: **19m 5s**

I run build process with `NODE_OPTIONS=--max-old-space-size=32000 gatsby build --verbose` 

I realise that its not a linear dependency I doubling cpu cores won't give me x2 boost. But Its a strange situation and I think that its a bottleneck somewhere in gatsbyjs core.

ps. memory usage doesn't exceeds 8GB

**Comment by KlausEverWalkingDev** [comment]
So you're looking for decreasing in build time and not increasing, right?

**Reply by snussik** [reply]
Sure, edited ) As on my production data source with 120k records 16cpu/32gb it builds 10 hours.

**Comment by ascorbic** [comment]
Hi. Are you able to share your repo, or at least the build log output? 5000 pages is not particularly large, so the bottleneck might be something simple.
